content/usgs-packages/sbtools_Get.Rmd

```diff
@@ -0,0 +1,179 @@
+
+First, let's inspect the parent item. The function to use is `item_get_parent`, and the item id is the only necessary argument.
+
+```{r sbtools-parent}
+ex_id <- "4f4e479de4b07f02db491e34"
+ex_id_parent <- item_get_parent(ex_id)
+ex_id_parent$title
+```
+
+Now, let's see if this item has any children by using the `item_list_children` function. Notice that this function says "list" and not "get" as the previous one did. Functions with "list" only return a few fields associated with each item. Functions with "get" are pulling down all available information, including files, associated with an item.
+
+```{r sbtools-children}
+ex_id_children <- item_list_children(ex_id)
+length(ex_id_children)
+sapply(ex_id_children, function(item) item$title)
+```
+
+Let's check to see if this item has any files attached to it using `item_list_files`. This will return a dataframe with the three columns: `fname` (filename), `size` (file size in bytes), and `url` (the URL to the file on ScienceBase).
+
+```{r sbtools-files}
+ex_id_files <- item_list_files(ex_id)
+nrow(ex_id_files)
+ex_id_files$fname
+```
+
+To actually get the files into R as data, you need to use their URLs and the appropriate parsing function. Both of the files returned for this item are XML, so you can use the `xml2` function, `read_xml`. As practice, we will download the first XML file.
+
+```{r sbtools-filedownload}
+xml2::read_xml(ex_id_files$url[1])
+```
+
+You can also inspect specific metadata fields of ScienceBase items. To do this, use the `item_get_fields` function. This function requires a second argument to the item id called `fields` that is a character vector of the fields you want to retrieve. See the [developer documentation for a SB item model](https://my.usgs.gov/confluence/display/sciencebase/ScienceBase+Item+Core+Model) for a list of potential fields. You can also use the argument `drop` to indicate that  if only one field is requested, the object returned remains a list (`drop=FALSE`) or becomes a vector (`drop=TRUE`). The default is `drop=TRUE`.
+
+```{r sbtools-fields}
+# request multiple fields
+multi_fields <- item_get_fields(ex_id, c("summary", "tags"))
+length(multi_fields)
+names(multi_fields)
+
+# single field, drop=TRUE
+single_field_drop <- item_get_fields(ex_id, "summary")
+names(single_field_drop)
+class(single_field_drop)
+
+# single field, drop=FALSE
+single_field <- item_get_fields(ex_id, "summary", drop=FALSE)
+single_field
+class(single_field)
+```
+
+If a field is empty, it will return `NULL`.
+
+```{r sbtools-fields-empty}
+# request a nonexistent fields
+item_get_fields(ex_id, c("dates", "citation"))
+```
+
+Now that we've inspected the item, let's actually pull the item down. There are a number of extra fields to inspect now.
+
+```{r sbtools-get}
+ex_id_item <- item_get(ex_id)
```

Comment:
Hmm...in light of all this re-education I've just had, what would you think about deleting both `as.sbitem` and `item_get_fields`? If `item_get_fields` is essentially a less-capable little sister of `item_get`...why would someone ever choose the function that requires you to dig around in documentation until they find the right field name, when they could just look through their complete item for it? What do you think?